File: src/main/java/client/inventory/operations/RemoveInventoryOperation.java
package client.inventory.operations;

import client.inventory.ItemInventoryType;
import client.inventory.ModifyInventoryOperationType;
import util.packet.PacketWriter;

public class RemoveInventoryOperation extends AbstractModifyInventoryOperation {

    public RemoveInventoryOperation(ItemInventoryType inventory, short slot) {
        super(inventory, slot);
    }

    @Override
    protected ModifyInventoryOperationType getType() {
        return ModifyInventoryOperationType.REMOVE;
    }

    @Override
    protected void encodeData(PacketWriter pw) {
        // keep empty
    }
}

File: src/main/java/client/inventory/operations/RemoveInventoryOperation.kt
package client.inventory.operations

import client.inventory.ItemInventoryType
import client.inventory.ModifyInventoryOperationType
import util.packet.PacketWriter

class RemoveInventoryOperation(inventory: ItemInventoryType, slot: Short) : AbstractModifyInventoryOperation(inventory, slot) {

    override val type = ModifyInventoryOperationType.REMOVE

    override fun encodeData(pw: PacketWriter) {
        // keep empty
    }
}